tests/proxy_test.go

```diff
@@ -117,14 +117,14 @@ func TestProxyHandleDialError_GRPC(t *testing.T) {
 	time.Sleep(time.Second)
 
 	// run test client
-	tunnel, err := client.CreateSingleUseGrpcTunnel(proxy.front, grpc.WithInsecure())
```

Comment:
Can we add a context timeout test for SingleUse?

Reply:
Done - but one of my two new tests isn't passing and I can't figure out why. (Is it the test itself or a remaining defect?)

Reply:
Filed https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/240 to get these tests merged. 
Would like to land these changes in K/K to confirm the fix, so not holding for these tests, yet...